src/connection.ts

```diff
@@ -3549,9 +3548,8 @@ Connection.prototype.STATE = {
               }, getTokenFromCredentials);
             } else if (authentication.type === 'azure-active-directory-msi-app-service') {
               loginWithAppServiceMSI({
-                msiEndpoint: authentication.options.msiEndpoint,
-                msiSecret: authentication.options.msiSecret,
-                resource: fedAuthInfoToken.spn
+                resource: fedAuthInfoToken.spn,
+                clientId: authentication.options.clientId
```

Comment:
Is there a specific reason for dropping the `msiEndpoint` and `msiSecret` options here? 🤔

Reply:
It doesn't appear to be needed according to the [docs](https://github.com/Azure/ms-rest-nodeauth) 
"`The code below works for both system managed and user-assigned managed identities. You can leave the options empty if you want to use system managed identity. If you want to use the user-assigned managed identity, you must at least provide the clientId in the options.`"

Reply:
That is correct, and that is what made user identities not work correctly here. But the `loginWithAppServiceMSI` still has support for the `msiEndpoint` and `msiSecret`, and I think we want to keep supporting those two options, in addition to `clientId`. 🙇‍♂️

Reply:
Ok 👍